Makefile

```diff
@@ -123,8 +123,8 @@ build-in-docker:
 build-minimal: $(BUILD_FOLDER)
 	CGO_ENABLED=0 go build -o $(BUILD_FOLDER)/assisted-service cmd/main.go
 
-build-image: build
-	docker build $(CONTAINER_BUILD_PARAMS) -f Dockerfile.assisted-service . -t $(SERVICE)
+build-image: build generate-dockerfiles
```

Comment:
I don't see a need to regenerate Dockerfiles on every image build. This would be fine to leave in it's own target

Reply:
Sorry for splitting the discussion, see my response in the other PR.

Reply:
The generation is extremely quick (much less than a second), it's a very simple jinja template. 

Regeneration on every build saves developers from making the mistake of changing the template and forgetting to generate. Any particular reason you think it should be avoided?

Reply:
It's defined as a wildcard "generate-*"

Reply:
Good point regarding the dependency

Reply:
For most development there isn't changes to the Dockerfiles. The case of regeneration is when you'd be making changes to the Dockerfiles and you would want to run the target separately because you will want to test those changes in isolation. Coupling this together makes it hard to reuse existing targets and testing your actual changes in isolation. Since most of the time it would be a noop I don't see the value in coupling it to a target that is used commonly. Testing changes in isolation is already difficult for the AI project too.

Reply:
Fair enough. In any case, I think [this PR](https://github.com/omertuc/assisted-service/pull/1) solves the issue much more elegantly than this PR, so no reason to continue this discussion.